graylog2-server/src/main/java/org/graylog2/bindings/NamedInjectConfigParametersModule.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.bindings;
+
+import com.github.joschi.jadconfig.Parameter;
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import com.google.inject.util.Providers;
+import org.reflections.ReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.github.joschi.jadconfig.ReflectionUtils.getAllFields;
+import static com.github.joschi.jadconfig.ReflectionUtils.getFieldValue;
+import static com.google.inject.name.Names.named;
+import static org.graylog2.shared.utilities.StringUtils.f;
+
+public class NamedInjectConfigParametersModule extends AbstractModule {
```

Comment:
What do you think about naming this just `ConfigParametersModule`?

Reply:
That name is derived from `com.github.joschi.jadconfig.guice.NamedConfigParametersModule`

How about `NamedConfigParametersOverrideModule`? 

However the name, I think some java doc would clarify what it does